litestar/openapi/plugins.py

```diff
@@ -0,0 +1,694 @@
+                  data-url="openapi.json">
+                </script>
+                <script src="{self.js_url}" crossorigin></script>
+                """
+
+        return f"""
+                <!DOCTYPE html>
+                    <html>
+                        {head}
+                        {body}
+                    </html>
+                """.encode()
+
+    def receive_router(self, router: Router) -> None:
+        """Receive the router that serves the OpenAPI UI.
+
+        Adds a route to serve the Scalar.com OpenAPI UI CSS
+        pulled from the Litestar GitHub branding repo.
+
+        Args:
+            router: The router that serves the OpenAPI UI.
+        """
+        if self.css_url == self._default_scalar_css_url:
+            router.register(
+                get("/_litestar-internal/scalar.css", media_type=MediaType.CSS, sync_to_thread=False)(
+                    self.render_scalar_css
+                ),
+            )
+
+    def render_scalar_css(self) -> bytes:
+        """Fetch and render the Scalar CSS file from GitHub.
+
+        Returns:
+            The CSS content as bytes.
+        """
+        try:
+            with httpx.Client() as client:
+                response = client.get(self._default_litestar_css_url)
+                response.raise_for_status()
+                return response.content
+        except httpx.HTTPError:
+            here = Path(__file__).parent
+            css_path = here / "styles/scalar.css"
+            return css_path.read_bytes()
+
+
+class StoplightRenderPlugin(OpenAPIRenderPlugin):
+    """Render an OpenAPI schema using StopLight Elements."""
+
+    def __init__(
+        self,
+        *,
+        version: str = "7.7.18",
+        js_url: str | None = None,
+        css_url: str | None = None,
+        path: str | Sequence[str] = "/elements",
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the OpenAPI UI render plugin.
+
+        Args:
+            version: StopLight Elements version to download from the CDN. If js_url is provided, this is ignored.
+            js_url: Download url for the StopLight Elements JS bundle. If not provided, the version will be used to
+                construct the url.
+            css_url: Download url for the StopLight Elements CSS bundle. If not provided, the version will be used to
+                construct the url.
+            path: Path to serve the OpenAPI UI at.
+            **kwargs: Additional arguments to pass to the base class.
+        """
+        self.js_url = js_url or f"https://unpkg.com/@stoplight/elements@{version}/web-components.min.js"
+        self.css_url = css_url or f"https://unpkg.com/@stoplight/elements@{version}/styles.min.css"
+        super().__init__(path=path, **kwargs)
+
+    def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
+        """Render an HTML page for StopLight Elements.
+
+        Notes:
+            - override this method to customize the template.
```

Comment:
```suggestion
       .. note:: Override this method to customize the template.
```
utilize sphinx admonition